src/viam/app/data/client.py

```diff
@@ -290,10 +320,41 @@ async def bounding_box_labels_by_filter(self, filter: Optional[Filter] = None) -
         response: BoundingBoxLabelsByFilterResponse = await self._data_client.BoundingBoxLabelsByFilter(request, metadata=self._metadata)
         return response.labels
 
-    # TODO(RSDK-3637): Implement
-    async def data_capture_upload(self, metadata: UploadMetadata, sensor_contents: Optional[List[SensorData]] = None) -> None:
-        raise NotImplementedError()
+    async def data_capture_upload(self, metadata: UploadMetadata , sensor_contents: List[SensorData]) -> None:
+        """Upload binary or tabular sensor data.
+
+        Sync binary or tabular data collected on a robot through a specific component (i.e. a motor) along with the relevant metadata with
+        app.viam.com.
+
+        Args:
+            metadata (viam.app.proto.datasync.UploadMetaData): Metadata relating the data collected with the corresponding robot part. Must
+                specify a part ID, component type, component name, method name, and data type.
+            sensor_contents (List[viam.app.proto.datasync.SensorData]): A list of the data to be uploaded. If the metadata specifies a
+                binary data type, the "binary" field of sensor_contents will be uploaded. Alternatively, the "struct" field will be uploaded
+                if the data type is tabular. Length of sensor_contents must be 1 if uploading a binary data capture. Otherwise, must simply
+                be nonempty.
+
+        Raises:
+            GRPCError: If a required data field is not defined.
+        """
+        request = DataCaptureUploadRequest(metadata=metadata, sensor_contents=sensor_contents)
+        _: DataCaptureUploadResponse = await self._data_sync_client.DataCaptureUpload(request, metadata=self._metadata)
 
-    # TODO(RSDK-3637): Implement
     async def file_upload(self, metadata: UploadMetadata, file_contents: Optional[FileData]) -> None:
-        raise NotImplementedError()
+        """Upload arbitrary file data.
+
+        Sync file data that may be stored on a robot along with the relevant metadata to app.viam.com.
+
+        Args:
+            metadata (viam.app.proto.datasync.UploadMetaData): Metadata for the file to be uploaded. Must specify a part ID.
```

Comment:
I think we don't care about part ID for file upload, but do care about file name and extension?

Reply:
But also see above about just asking for what we need and making an `UploadMetadata` under the hood.

Reply:
I also thought that we wouldn't need a part ID for this method, but when I run the function with either an invalid part id (or just without one), I receive a GRPCError that says (no robot part found). I haven't been able to to find where this is coming from in app.. I've only been able to find the string "no robot part found" in tests in app, so maybe I'll reach out to Alexa again and see if something else is up.

Reply:
Okay I found where the error is coming up, a valid id is definitely required and used in this request.